src/statement/dataTypes.ts

```diff
@@ -34,28 +34,30 @@ const typeMapping = {
   bytea: "bytea"
 };
 
+const COMPLEX_TYPE = /^(nullable|array)\((.+)\)( null)?/;
+const STRUCT_TYPE = /^(struct)\((.+)\)/;
+const NULLABLE_TYPE = /^(.+)( null)$/;
+const DATETIME_TYPE = /datetime64(.+)/i;
+const TIMESTAMP_TYPE = /timestamp_ext(.+)/i;
+const DECIMAL_TYPE = /decimal(.+)/i;
+const NUMERIC_TYPE = /numeric(.+)/i;
+
 const getMappedType = (innerType: string) => {
-  const type = typeMapping[innerType as keyof typeof typeMapping];
-  if (type) {
-    return type;
+  const match = NULLABLE_TYPE.exec(innerType);
+  const type = match ? match[1] : innerType;
+  const nullableSuffix = match ? " null" : "";
+  const mappedType = typeMapping[type as keyof typeof typeMapping];
+  if (mappedType) {
+    return `${mappedType}${nullableSuffix}`;
   }
-  if (
-    RegExp(/datetime64(.+)/i).exec(innerType) ||
-    RegExp(/timestamp_ext(.+)/i).exec(innerType)
-  ) {
-    return typeMapping.timestamp;
+  if (RegExp(DATETIME_TYPE).exec(type) || RegExp(TIMESTAMP_TYPE).exec(type)) {
+    return `${typeMapping.timestamp}${nullableSuffix}`;
   }
-  if (
-    RegExp(/decimal(.+)/i).exec(innerType) ||
-    RegExp(/numeric(.+)/i).exec(innerType)
-  ) {
-    return typeMapping.decimal;
+  if (RegExp(DECIMAL_TYPE).exec(type) || RegExp(NUMERIC_TYPE).exec(type)) {
+    return `${typeMapping.decimal}${nullableSuffix}`;
   }
```

Comment:
Do we return null if we don't recognize the type? Let's add the explicit `return null` just for clarity